Tasks/Common/ios-signing-common/ios-signing-common.ts

```diff
@@ -602,8 +598,83 @@ export async function getP12CommonName(p12Path: string, p12Pwd: string) {
     return commonName;
 }
 
-export async function deleteCert(keychainPath: string, certSha1Hash: string) {
+export async function deleteCert(keychainPath: string, certSha1Hash: string): Promise<void> {
     let deleteCert: ToolRunner = tl.tool(tl.which('security', true));
     deleteCert.arg(['delete-certificate', '-Z', certSha1Hash, keychainPath]);
     await deleteCert.exec();
 }
+
+export async function getP12PrivateKeyName(p12Path: string, p12Pwd: string): Promise<string> {
+    //openssl pkcs12 -in <p12Path> -nocerts -passin pass:"<p12Pwd>" -passout pass:"<p12Pwd>" | grep 'friendlyName'
+    //We pipe through grep so we we don't log the private key, even encrypted.
+    tl.debug('getting the P12 private key name');
+    let opensslPath: string = tl.which('openssl', true);
+    let openssl: ToolRunner = tl.tool(opensslPath);
+    if (!p12Pwd) {
+        // if password is null or not defined, set it to empty
+        p12Pwd = '';
+    }
+    openssl.arg(['pkcs12', '-in', p12Path, '-nocerts', '-passin', 'pass:' + p12Pwd, '-passout', 'pass:' + p12Pwd]);
+
+    let grepPath: string = tl.which('grep', true);
+    let grep: ToolRunner = tl.tool(grepPath);
+    grep.arg(['friendlyName']);
+    openssl.pipeExecOutputToTool(grep);
+
+    let privateKeyName: string;
+    openssl.on('stdline', function (line: string) {
```

Comment:
`stdline` <= prevents shearing?

Reply:
Yup. Unfortunately there's no mocking for `stdline` (https://github.com/Microsoft/vsts-task-lib/issues/227). I'm switching back to `stdout` to get the tests passing. I hope to find time to fix the mocking and then return to iOS signing and switch over to `stdline` everywhere.